src/satosa/micro_services/custom_routing.py

```diff
@@ -2,14 +2,117 @@
 from base64 import urlsafe_b64encode
 
 from satosa.context import Context
+from satosa.internal import InternalData
+
 from .base import RequestMicroService
 from ..exception import SATOSAConfigurationError
 from ..exception import SATOSAError
+from ..exception import SATOSAStateError
 
 
 logger = logging.getLogger(__name__)
 
 
+class CustomRoutingError(SATOSAError):
+    """
+    SATOSA exception raised by CustomRouting rules
+    """
+    pass
+
+
+class DecideBackendByTarget(RequestMicroService):
+    """
+    Select which backend should be used based on who is the SAML IDP
+    """
+
+    def __init__(self, config, *args, **kwargs):
+        """
+        Constructor.
+        :param config: microservice configuration loaded from yaml file
+        :type config: Dict[str, Dict[str, str]]
+        """
+        super().__init__(*args, **kwargs)
+        self.target_mapping = config['target_mapping']
+        self.endpoint_paths = config['endpoint_paths']
+        self.default_backend = config['default_backend']
+
+        if not isinstance(self.endpoint_paths, list):
```

Comment:
`endpoint_paths` is required to be a list, but the example shows a single value `".*/disco"`

Reply:
Fixed here
https://github.com/peppelinux/SATOSA/blob/DecideBackendByTarget/example/plugins/microservices/target_based_routing.yaml.example

Fixed here: https://github.com/peppelinux/SATOSA/commit/4e21146401503a3f4340fedab905de668aa7dd9f